lib/util/stream.js

```diff
@@ -106,11 +106,50 @@ class PartialPipe {
   pipe(out) { return reduce(((x, y) => x.pipe(y)), this.streams[0], this.streams.slice(1)).pipe(out); }
 }
 
+// allWithStreams() is very similar to Promise.all() and returns the same
+// result. The key difference is that if `values` contains a stream or a promise
+// that resolves to a stream, then the fate of the stream will be tied to the
+// fate of each promise within `values`. If any promise is rejected, then the
+// stream will be immediately destroyed.
+const allWithStreams = (values) => {
+  // If/when a promise is rejected, we will need to destroy any streams that we
+  // have already seen (streamsBeforeError), as well as remember to destroy any
+  // streams that we see later (anyError).
+  let anyError = false;
+  const streamsBeforeError = [];
+  const handleStream = (stream) => {
+    if (anyError)
+      stream.destroy();
+    else
+      streamsBeforeError.push(stream);
+  };
+
+  for (const value of values) {
+    Promise.resolve(value)
```

Comment:
We usually call `resolve()` from lib/util/promise.js rather than call `Promise.resolve()` directly. However, `resolve()` isn't otherwise used in this file, and importing it resulted in ESLint errors: there are several variables in this file that are also named `resolve`. It feels preferable to use `Promise.resolve()` here rather than rename those variables or alias `resolve()`.